src/services/filterPeople.ts

```diff
@@ -0,0 +1,44 @@
+import { Person } from '../types';
+
+export function filterPeople(
+  genderType: string | null,
+  centuries: string[],
+  queryType: string | null,
+  people: Person[],
+) {
+  function matchingQuery(person: Person, query: string) {
+    const { name, motherName, fatherName } = person;
+    const normalizedName = name.toLowerCase();
+
+    const normalizedMotherName = motherName?.toLowerCase();
+    const normalizedFatherName = fatherName?.toLowerCase();
+
+    if (
+      normalizedName.includes(query) ||
+      normalizedMotherName?.includes(query) ||
+      normalizedFatherName?.includes(query)
+    ) {
+      return name;
+    } else {
+      return;
```

Comment:
The `matchingQuery` function returns `name` if a match is found, but returns `undefined` otherwise. This is fine for filtering purposes, but it might be clearer to return a boolean value (`true` or `false`) to indicate a match, which would make the intention of the function more explicit.